app/models/manageiq/providers/ansible_tower/shared/inventory/collector/target_collection.rb

```diff
@@ -0,0 +1,66 @@
+module ManageIQ::Providers::AnsibleTower::Shared::Inventory::Collector::TargetCollection
+  def connection
+    @connection ||= manager.connect
+  end
+
+  def initialize(_manager, _target)
+    super
+  end
+
+  def inventories
+    find_records(:inventory_root_groups, connection.api.inventories)
+  end
+
+  def hosts
+    find_records(:configured_systems, connection.api.hosts)
+  end
+
+  def job_templates
+    find_records(:configuration_scripts, connection.api.job_templates)
+  end
+
+  def configuration_workflows
+    find_records(:configuration_scripts, connection.api.workflow_job_templates)
+  end
+
+  def projects
+    find_records(:configuration_script_sources, connection.api.projects)
+  end
+
+  def credentials
+    find_records(:credentials, connection.api.credentials)
+  end
+
+  protected
+
+  # Calls API and finds existing records.
+  # @param inventory_collection_name [Symbol] IC name (as identified in persister's definitions)
+  # @param endpoint - endpoint for AnsibleTowerClient api call
+  def find_records(inventory_collection_name, endpoint)
+    refs = references(inventory_collection_name)
+    return [] if refs.blank?
+
+    if refs.size == 1
+      begin
+        [endpoint.find(refs[0])]
+      rescue AnsibleTowerClient::ResourceNotFoundError
+        [nil]
+      end
+    else
+      endpoint.all(:id__in => refs.join(','))
```

Comment:
check if there is a limit for this, e.g. for amazon, you can search for max 200 ids. Then you can do something like the multi_query here https://github.com/Ladas/manageiq-providers-amazon/blob/4197be8d1dee08a18c87483cf5ada03bebb6f5c1/app/models/manageiq/providers/amazon/inventory/collector/target_collection.rb#L23 , which is just an iterator

Reply:
@Ladas hm, I didn't find it in documentation so I need to make additional spec